src/components/Layout/Header/NavBar/ChainMenu.tsx

```diff
@@ -0,0 +1,113 @@
+import { ChevronDownIcon } from '@chakra-ui/icons'
+import { Menu, MenuButton, MenuGroup, MenuItem, MenuList } from '@chakra-ui/menu'
+import { Box, Button, Flex, useColorModeValue } from '@chakra-ui/react'
+import { avalancheChainId, ethChainId, fromChainId } from '@shapeshiftoss/caip'
+import { bnOrZero } from '@shapeshiftoss/investor-foxy'
+import { useEffect, useMemo, useState } from 'react'
+import { AssetIcon } from 'components/AssetIcon'
+import { CircleIcon } from 'components/Icons/Circle'
+import { getChainAdapters } from 'context/PluginProvider/PluginProvider'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { selectAssetById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+// TODO: This should move to lib, we should be able to get `getName()` on each adapter
```

Comment:
to be really explicit, this should be like `getDisplayName()` so it's obvious it's only for view layer purposes vs `getName()` *might* be interpreted as like an id